### PR TITLE
[CI-only] Support UBI images

### PR DESCRIPTION
**Pull request by mdeggies**
### Description
This migrates the docker target and docker entrypoint used to build the UBI image from https://github.com/hashicorp/docker-consul/tree/master/ubi into consul. The UBI image will be built along with other artifact types in github actions on merges to `main`, and will be released alongside the other consul docker images via CRT.

### Testing & Reproduction steps
* Tested a build on this branch by adding the branch name to the allow-list in `build.yml`. Link to green build: https://github.com/hashicorp/consul/actions/runs/2385840986
* Tested the UBI image locally with the following steps:
1. Download the ubi tarball `consul_ubi_linux_amd64_1.12.0-dev_b2683af13e34879895a59c54e43a0a5c30f88128.docker.redhat.tar` from github artifacts available here: https://github.com/hashicorp/consul/actions/runs/2385840986
2. Run `docker load -i consul_ubi_linux_amd64_1.12.0-dev_b2683af13e34879895a59c54e43a0a5c30f88128.docker.redhat.tar`
3. Verify the image is available and grab the ID: `docker images | grep ubi`
4. Run smoke tests against image to verify the container is starting appropriately: `docker run -it 96c079fcb194`
5. Run smoke test to verify commit sha is being displayed properly in `version`: ```$ docker run --rm -it --entrypoint bash 96c079fcb194
bash-4.4$ consul --version
Consul v1.12.0-dev
Revision b2683af1
Protocol 2 spoken by default, understands 2 to 3 (agent will automatically use protocol >2 when speaking to compatible agents)```
6. Will need an engineer to validate other commands using the image to ensure everything is set up properly. Thanks in advance! 

### Next steps to follow 
* Add backport labels for 1.10.x, 1.11.x, 1.12.x
* On the opened OSS->ENT PR's, we'll need to update the redhat project ID to reflect the enterprise project ID. This is the line that will need to be updated: https://github.com/hashicorp/consul/blob/80cda3b986da5f319042aee44895ae5a1c8abb81/.github/workflows/build.yml#L266. The project ID for ENT (available publicly at https://catalog.redhat.com/software/containers/hashicorp/consul-enterprise/612d0116eb34b736524936b2) is `612d0116eb34b736524936b2`. 
* On the opened OSS->ENT PR's, add logic in the Dockerfile to place the EULA/Terms of Eval (same as we're doing for vault-ent here: https://github.com/hashicorp/vault-enterprise/pull/2869)

### Links
Link to the documentation on UBI image building/publishing via CRT: https://go.hashi.co/crt-ubi-images

### CODE REVIEW

**Review by david-yu — APPROVED** [review]

**Comment by david-yu** [comment]
https://github.com/hashicorp/docker-consul/blob/master/ubi/Dockerfile which was the original UBI Docker image file for reference.

File: Dockerfile [hunk 1]
```diff
@@ -188,6 +188,99 @@ COPY .release/docker/docker-entrypoint.sh /usr/local/bin/docker-entrypoint.sh
...
 
+# By default you'll get an insecure single-node development server that stores
+# everything in RAM, exposes a web UI and HTTP endpoints, and bootstraps itself.
+# Don't use this configuration for production.
+CMD ["agent", "-dev", "-client", "0.0.0.0"]
+
+
+# Red Hat UBI-based image
+# This target is used to build a Consul image for use on OpenShift.
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.5 as ubi
```
Comment (david-yu) [inline Dockerfile:199]:
```suggestion
FROM registry.access.redhat.com/ubi8/ubi-minimal:8.6 as ubi
```

Comment (david-yu) [reply Dockerfile:199]: Current image is 8.6

**Comment by mdeggies** [comment]
OK, going to merge this in and resolve any ENT conflicts.

**Comment by mdeggies** [comment]
OSS->ENT automation was turned off, so for now I'll open the ENT PR's manually.. will link here once they're ready

**Comment by mdeggies** [comment]
ENT PR's:
`main`: https://github.com/hashicorp/consul-enterprise/pull/1959
`release/1.10.x`: https://github.com/hashicorp/consul-enterprise/pull/1962
`release/1.11.x`: https://github.com/hashicorp/consul-enterprise/pull/1963
`release/1.12.x`: https://github.com/hashicorp/consul-enterprise/pull/1964

**Comment by mdeggies** [comment]
All PR's merged.

**Pull request merged**